controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -123,16 +124,12 @@ func (r *HostFirmwareSettingsReconciler) Reconcile(ctx context.Context, req ctrl
 	hfs := &metal3v1alpha1.HostFirmwareSettings{}
 	info := &rInfo{log: reqLogger, hfs: hfs, bmh: bmh}
 	if err = r.Get(ctx, req.NamespacedName, hfs); err != nil {
+		// The HFS resource may have been deleted
 		if k8serrors.IsNotFound(err) {
-			// A resource doesn't exist, create one
-			if err = r.newHostFirmwareSettings(info, req.NamespacedName); err != nil {
-
-				return ctrl.Result{}, errors.Wrap(err, "could not create firmware settings")
-			}
-		} else {
-			// Error reading the object - requeue the request.
-			return ctrl.Result{}, errors.Wrap(err, "could not load host firmware settings")
+			return ctrl.Result{}, errors.Wrap(err, "hostFirmwareSettings not found")
```

Comment:
Note that returning an error will cause an immediate reconciliation (rate limited). Considering that we'll have to wait the next BMH reconciliation loop for recreating the missing resource, maybe this case could be treated not as an error but as a waiting condition (with `RequeueAfter` set to a reasonable value, ie `provisionerRetryDelay`)

Reply:
Changed to return requeue time